PSReadLine/PlatformWindows.cs

```diff
@@ -614,4 +611,350 @@ public override void BlankRestOfLine()
+    {
+        var procsToTerminate = new List<Process>();
+
+        // The theory for why more than one round might be needed is that the same race
+        // between process creation and console attachment that could cause lingering
+        // processes in the first place could cause us to need a second round of
+        // cleanup... but I've never actually seen more than one round be needed. Probably
+        // because in my specific scenario the process that was spawning processes got
+        // taken out with the original ctrl+c signal.
+        //
+        // If it takes more than a few rounds of cleanup, we may be in some kind of
+        // pathological situation, and we'll bow out.
+        int round = 0;
+        int killAttempts = 0;
+
+        while ((round++ < MaxRounds) &&
+               (GatherStragglers(procsToTerminate) > 0))
+        {
+            // We'll give them up to GracePeriodMillis for them to exit on their
+            // own, in case they actually did receive the original ctrl+c, and are
+            // just a tad slow shutting down.
+            ulong deadline = GetTickCount64() + (ulong) DefaultGraceMillis;
+
+            var notDeadYet = procsToTerminate.Where(
+                    (p) => !p.WaitForExit(MillisLeftUntilDeadline(deadline)));
+
+            foreach (var process in notDeadYet)
+            {
+                try
+                {
+                    killAttempts++;
+                    process.Kill();
+                }
+                // Ignore problems; maybe it's gone already, maybe something else;
+                // whatever.
+                catch (InvalidOperationException) { }
+                catch (Win32Exception) { }
+            }
+
+            foreach (var process in procsToTerminate)
+            {
+                process.Dispose();
+            }
+        } // end retry loop
+
+        // In forcible termination scenarios, if there was a child updating the terminal's
+        // progress state, it may be left stuck that way... we can clear that out.
+        if (ClearProgress)
+        {
+            uint consoleMode = GetConsoleOutputMode();
+            if (ItLooksLikeWeAreInTerminal())
+            {
+                // We can use the [semi-]standard OSC sequence:
+                // https://conemu.github.io/en/AnsiEscapeCodes.html#ConEmu_specific_OSC
+                if (0 != (consoleMode & (uint) ENABLE_VIRTUAL_TERMINAL_PROCESSING))
+                {
+                    // Use "bell" if we actually tried to whack anything.
+                    char final = (killAttempts > 0) ? '\a' : '\x001b';
+                    Console.Write("\x001b]9;4;0;0" + final);
+                }
+            }
+            else
+            {
+                IntPtr hwnd = GetConsoleWindow();
+                if (hwnd != IntPtr.Zero)
+                {
+                    int ret = TaskbarProgress.SetProgressState(hwnd, TaskbarStates.NoProgress);
```

Comment:
1: I tried 7.2.13, 7.3.6, and 7.4.0-preview.4, in both Terminal and conhost. I added a `Console.WriteLine` to print out the `ret` value, and it was always 0 (and progress was cleared), so it seems to work. 🤷

2: both the VT sequence and the COM API are two different paths to the same thing; so we do VT if we believe it will work, else we fall back to COM. Make sense?